Backend/Gmfctn/Controllers/AchievementController.cs

```diff
@@ -98,25 +120,21 @@ public async Task<ActionResult> DeleteAchievement(Guid Id, CancellationToken Can
                 throw Exc;
             }
         }
-        [HttpPut("{id}")]
+        [HttpPut("{Id}")]
+        [Authorize(Roles = "Admin")]
         public async Task<ActionResult> UpdateAchievement(Guid Id, AchievementUpdateDTO Achievement, CancellationToken Cancel)
         {
             try
             {
-                if (!ModelsValidator.AchievementIsValid(_Mapper.Map<AchievementCreateDTO>(Achievement)))
-                    ModelState.AddModelError("achievement", "Contains errors in parametrs.");
-                if (ModelState.IsValid)
+                if (await AchievementService.UpdateAchievement(Id, Achievement, Cancel))
                 {
-                    var _Achievement = await UnitOfWork.AchievementRepository.DbSet.FirstOrDefaultAsync(item => item.Id == Id);
-                    _Mapper.Map(Achievement, _Achievement);
-                    UnitOfWork.AchievementRepository.Update(_Achievement); 
-                    await UnitOfWork.SaveChangesAsync(Cancel);
                     return Ok();
                 }
-                else 
-                {
-                    return BadRequest(ModelState);
-                }
+                return BadRequest();
```

Comment:
The same as above about handling errors and return from Service logic

Reply:
Fixed